providers/azureblob/src/main/java/org/jclouds/azure/storage/filters/SharedKeyLiteAuthentication.java

```diff
@@ -85,23 +91,95 @@ public SharedKeyLiteAuthentication(SignatureWire signatureWire,
       this.crypto = crypto;
       this.utils = utils;
       this.signatureWire = signatureWire;
+	  this.storageUrl = URI.create("https://" + creds.get().identity + ".blob.core.windows.net/");
       this.creds = creds;
+	  this.identity = creds.get().identity;
+	  this.credential = creds.get().credential;
       this.timeStampProvider = timeStampProvider;
+	  this.sasAuthentication = false;
    }
-
+	/** 
+	* this is an updated filter method, which decides whether the SAS or SharedKeyLite 
+	* is used and applies the right filtering.  
+	* 
+	*/
    public HttpRequest filter(HttpRequest request) throws HttpException {
+	 this.sasAuthentication = authSAS(this.credential);
+	 if (this.sasAuthentication){
+		request = filterSAS(request, this.credential);
+	 } else {
+		request = filterKey(request);
+	 }
+	 utils.logRequest(signatureLog, request, "<<");
+     return request;
+   }
+   
+    /**
+    * 
+    * this method checks the length of the authentication string, and decides, which auth method it represents. 
+    * 
+    */
+   public boolean authSAS(String credential) {
+	   int length = credential.length();
+	   
+	   boolean ruleSAS = (length < 150) && (length > 120);
+	   boolean ruleKey = (length < 100) && (length > 70 );
+	   
+	   if (ruleSAS) {
+		   return true;
+	   } else if (ruleKey) {
+		   return false;
+	   } else {
+		   return false;
+	   }
+   }
+	/** 
+	* this filter method is applied only for the cases with SAS Authentication. 
+	* 
+	*/
+	public HttpRequest filterSAS(HttpRequest request, String credential) throws HttpException {
+		String containerName = "";
+		String blobName = "";
+		String requestLine = request.getRequestLine();
+		String[] containerBlob = cutUri(requestLine);
+		if (containerBlob[0] != null) {
+			containerName = containerBlob[0];
+		} 
+		if (containerBlob[1] != null) {
+			blobName = containerBlob[1];
+		}
+		request = request.toBuilder().endpoint(Uris.uriBuilder(storageUrl).appendPath(containerName).appendPath(blobName).query(credential).build()).build();
+		request = replaceDateHeader(request);
+		request = removeAuthorizationHeader(request);
```

Comment:
Prefer a builder pattern, e.g.,

```java
return request.toBuilder().endpoint(Uris.uriBuilder(storageUrl).appendPath(containerName).appendPath(blobName).query(credential).build()).build()
    replaceDateHeader(request)
    removeAuthorizationHeader(request);

Reply:
replaceDateHeader(HttpRequest) and removeAuthorizationHeader(request) both belong to ShareKeyLiteAuthentication class, not the HttpMessage Builder class. 
The original code is written the same way as I did initially: 
```
 public HttpRequest filter(HttpRequest request) throws HttpException {
      request = replaceDateHeader(request);
      String signature = calculateSignature(createStringToSign(request));
      request = replaceAuthorizationHeader(request, signature);
      utils.logRequest(signatureLog, request, "<<");
      return request;
   } 
```